### ISSUE TITLE
CORS error

### ISSUE THREAD

**Issue by Akshay-akkay** [apivideo/api.video-browser-to-rtmp#4]
**Describe the bug**
Access to XMLHttpRequest at 'http://localhost:1234/socket.io/?EIO=4&transport=polling&t=OLsh1L1' from origin 'http://localhost:52218' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

## Backend Code
```/// Import HTTP module
const http = require('http');

// '@api.video/browser-to-rtmp-server';
const BrowserToRtmpServer = require('@api.video/browser-to-rtmp-server');

/// Create HTTP server
const server = http.createServer();

/// Setup CORS

const browserToRtmpClient = new BrowserToRtmpServer(server, {
  cors: {
    origin: 'http://localhost:52218',
    methods: ['GET', 'POST'],
    credentials: true,
  },
});
server.listen(1234, () => {
  console.log('Server is running on port 1234');
});
```

**To Reproduce**
1. Setup Example Project
2. Start streaming from frontend
3. See console in chrome.

**Expected behavior**
Stream should start

**Screenshots**
<img width="540" alt="Screenshot 2023-01-03 at 1 38 17 PM" src="https://user-images.githubusercontent.com/66077920/210319701-c2abfacc-fcdc-4dd5-9a83-24b05a2c697a.png">

**Environment (please complete the following information):**
 - OS: macos
 - Browser  chrome
 - Version latest

**Comment by olivier-lando** [comment]
Hi @Akshay-akkay ,

Try to put the cors settings in the `socketio` value of the options, like in the [example](https://github.com/apivideo/api.video-browser-to-rtmp/blob/main/example/index.ts).

```javascript
const browserToRtmpServer = new BrowserToRtmpServer(server, {
  // ...
  socketio: {
    cors: {
      origin: "*",
      methods: ["GET", "POST"],
      credentials: true
    },
  },
  // ...
});
```

Please let me know if that solves your issue.

**Comment by Akshay-akkay** [comment]
Yes, I looked into the source code. And was able to resolve this. 

But now I am facing another issue namely, 

Unhandled Error (random_uid) 

on server side.

**Comment by olivier-lando** [comment]
Can you send the full error ?

**Comment by Akshay-akkay** [comment]
Thanks, for helping.

Here is the full error log - 
```Server is running on port 1234
Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('3a700d95-4d70-41be-80b8-edc6691767c3')
    at new NodeError (node:internal/errors:387:5)
    at BrowserToRtmpServer.emit (node:events:502:17)
    at BrowserToRtmpServer.emit (node:domain:482:12)
    at Connection.<anonymous> (/Users/akshay/Documents/GitHub/hhu_node_stream_proxy/node_modules/@api.video/browser-to-rtmp-server/src/index.ts:111:12)
    at Connection.emit (node:events:513:28)
    at Connection.emit (node:domain:482:12)
    at Connection.sendErrorToClient (/Users/akshay/Documents/GitHub/hhu_node_stream_proxy/node_modules/@api.video/browser-to-rtmp-server/src/connection.ts:106:10)
    at Connection.validateFfmpegConfig (/Users/akshay/Documents/GitHub/hhu_node_stream_proxy/node_modules/@api.video/browser-to-rtmp-server/src/connection.ts:120:12)
    at Connection.onStart (/Users/akshay/Documents/GitHub/hhu_node_stream_proxy/node_modules/@api.video/browser-to-rtmp-server/src/connection.ts:83:15)
    at Socket.<anonymous> (/Users/akshay/Documents/GitHub/hhu_node_stream_proxy/node_modules/@api.video/browser-to-rtmp-server/src/connection.ts:40:55) {
  code: 'ERR_UNHANDLED_ERROR',
  context: '3a700d95-4d70-41be-80b8-edc6691767c3'

}
```



<img width="1410" alt="Screenshot 2023-01-03 at 3 25 21 PM" src="https://user-images.githubusercontent.com/66077920/210335167-0ae73b35-cc5f-444f-ba92-7b019a976e2f.png">

**Comment by olivier-lando** [comment]
Can you check in your browser console if you can see any error?

**Comment by Akshay-akkay** [comment]
On browser getting only, Connection Refused Error

<img width="884" alt="Screenshot 2023-01-03 at 5 33 14 PM" src="https://user-images.githubusercontent.com/66077920/210353778-767d8a1e-e919-4f0b-8373-33ede26cc15d.png">

**Comment by olivier-lando** [comment]
The stack trace indicates that the error is thrown from `validateFfmpegConfig`. It looks like if the parameter defined on the client site were incorrect. Can you share the code snippet from the frontend that is responsible for instanciating `BrowserToRtmpClient` and calling the `start()` method ?
